src/Test/ImplicitMethodsIntegrationTest.php

```diff
@@ -41,49 +43,105 @@
-                    return new Stream('php://temp', 'rw');
-                }
-            ),
+            [RequestMethod::METHOD_HEAD, RequestMethod::METHOD_POST],
+        ];
+
+        yield 'HEAD: head, get' => [
+            RequestMethod::METHOD_HEAD,
+            [RequestMethod::METHOD_HEAD, RequestMethod::METHOD_GET],
+        ];
+
+        yield 'HEAD: post, head' => [
+            RequestMethod::METHOD_HEAD,
+            [RequestMethod::METHOD_POST, RequestMethod::METHOD_HEAD],
+        ];
+
+        yield 'HEAD: get, head' => [
+            RequestMethod::METHOD_HEAD,
+            [RequestMethod::METHOD_GET, RequestMethod::METHOD_HEAD],
         ];
-        yield RequestMethod::METHOD_OPTIONS => [
+
+        yield 'OPTIONS: options, post' => [
             RequestMethod::METHOD_OPTIONS,
-            new ImplicitOptionsMiddleware(
-                function () {
-                    return new Response();
-                }
-            ),
+            [RequestMethod::METHOD_OPTIONS, RequestMethod::METHOD_POST],
+        ];
+
+        yield 'OPTIONS: options, get' => [
+            RequestMethod::METHOD_OPTIONS,
+            [RequestMethod::METHOD_OPTIONS, RequestMethod::METHOD_GET],
+        ];
+
+        yield 'OPTIONS: post, options' => [
+            RequestMethod::METHOD_OPTIONS,
+            [RequestMethod::METHOD_POST, RequestMethod::METHOD_OPTIONS],
+        ];
+
+        yield 'OPTIONS: get, options' => [
+            RequestMethod::METHOD_OPTIONS,
+            [RequestMethod::METHOD_GET, RequestMethod::METHOD_OPTIONS],
         ];
     }
 
     /**
      * @dataProvider method
      */
-    public function testExplicitRequest(string $method, MiddlewareInterface $middleware)
+    public function testExplicitRequest(string $method, array $routes)
     {
-        $middleware1 = $this->prophesize(MiddlewareInterface::class)->reveal();
-        $middleware2 = $this->prophesize(MiddlewareInterface::class)->reveal();
-        $route1 = new Route('/api/v1/me', $middleware1, [$method]);
-        $route2 = new Route('/api/v1/me', $middleware2, [RequestMethod::METHOD_GET]);
-
+        $implicitRoute = null;
         $router = $this->getRouter();
-        $router->addRoute($route1);
-        $router->addRoute($route2);
+        foreach ($routes as $routeMethod) {
+            $route = new Route(
+                '/api/v1/me',
+                $this->prophesize(MiddlewareInterface::class)->reveal(),
+                [$routeMethod]
+            );
+            $router->addRoute($route);
+
+            if ($routeMethod === $method) {
+                $implicitRoute = $route;
+            }
+        }
+
+        $pipeline = new MiddlewarePipe();
+        $pipeline->pipe(new PathBasedRoutingMiddleware($router));
+        $pipeline->pipe(
+            $method === RequestMethod::METHOD_HEAD
+                ? $this->getImplicitHeadMiddleware($router)
+                : $this->getImplicitOptionsMiddleware()
+        );
```

Comment:
As this is an integration test, pipe both, in the expected order.

Additionally, in the patch I was working on yesterday, I created a `$responseFactory` that I composed into middleware that _should not_ return a generated response that looked like this:

```php
$responseFactory = function () {
    Assert::fail('Response generated when it should not be.');
};
```

This approach was what allowed me to determine why middleware order mattered here.

In this particular test case, I'd use that with the method not allowed middleware, and whichever implicit middleware represents the method _not_ being tested.